test.sh

```diff
@@ -24,12 +30,92 @@ function prepareOutputDir() {
   mkdir -p output
 }
 
-function createTestEnvironment() {
+# Build DAP test container & start the cluster
+function createDAPTestEnvironment() {
+  # docker-compose build --pull client-dap cuke-master postgres test-dap
+  docker-compose up -d client-dap cuke-master postgres test-dap
+
+  # Delay to allow time for conjur to come up
+  echo 'Waiting for conjur server to be healthy'
+  docker-compose run --rm test ./wait_for_server.sh
+}
+
+function loadDAPTestPolicy() {
+  echo '-----------------------test.sh------------------------------'
+  echo "Loading DAP test policy"
+  echo '------------------------------------------------------------'
+
+  # get DAP container id
+  dap_client_cid=$(docker-compose ps -q client-dap)
+
+  ssl_cert=$(docker-compose exec cuke-master cat /opt/conjur/etc/ssl/conjur.pem)
+  export CONJUR_SSL_CERTIFICATE=$(docker-compose exec cuke-master cat /opt/conjur/etc/ssl/conjur.pem)
+
+  docker-compose up -d client-dap
+  docker-compose exec client-dap conjur authn login -u admin -p secret
+
+  # copy test-policy into a /tmp/test-policy within the client container
+  docker cp test-policy ${dap_client_cid}:/tmp
+
+  docker-compose exec client-dap bash -c "conjur policy load root /tmp/test-policy/root.yml"
+}
+
+function initializeDAPCert () {
+  dap_client_cid=$(docker-compose ps -q client-dap)
+    # get the pem file from conjur server
+  CONJUR_ACCOUNT="cucumber"
+  CONJUR_PROXY="https://cuke-master"
+
+#  NOT WORKING
+  echo "remove old pem file"
+  rm -rf /test-cert/*
+
+  echo "fetch pem file from proxy https server"
+  exec_command='echo yes | conjur init -u '${CONJUR_PROXY}' -a '${CONJUR_ACCOUNT}' > tmp.out 2>&1'
+  docker exec ${dap_client_cid} /bin/bash -c "$exec_command"
+
+  echo "print command output"
+  print_command="cat tmp.out"
+  docker exec ${dap_client_cid} ${print_command}
+
+  echo "convert pem to der file and copy it to share memory"
+  convert_command="openssl x509 -outform der -in /root/conjur-cucumber.pem -out /test-cert/conjur-cucumber.der"
+  docker exec ${dap_client_cid} ${convert_command}
```

Comment:
Why do we have a different mixture of `docker exec` in some of these commands throughout this file instead of `docker-compose exec`? Some are needed but some are not so see if you can figure out why that might be the case.

Reply:
This is still relevant for the new code added - don't worry about the old code unless you want to fix that up too.